test/it/uk/gov/hmrc/exports/migrations/changelogs/submission/AddNotificationSummariesToSubmissionsISpec.scala

```diff
@@ -0,0 +1,325 @@
+/*
+ * Copyright 2022 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.exports.migrations.changelogs.submission
+
+import org.bson.Document
+import uk.gov.hmrc.exports.base.IntegrationTestMigrationToolSpec
+import uk.gov.hmrc.exports.migrations.changelogs.submission.AddNotificationSummariesToSubmissionsISpec._
+
+import scala.collection.JavaConverters._
+
+class AddNotificationSummariesToSubmissionsISpec extends IntegrationTestMigrationToolSpec {
+
+  override val collectionUnderTest = "submissions"
+  override val changeLog = new AddNotificationSummariesToSubmissions()
+
+  def prepareNotificationCollection(parsedNotifications: List[Document]): Boolean = {
+    val collection = getCollection("notifications")
+    removeAll(collection)
+    collection.insertMany(parsedNotifications.asJava).wasAcknowledged
+  }
+
+  "AddNotificationSummariesToSubmissions" should {
+
+    "update a Submission document" when {
+      "the document has a 'mrn' field and" when {
+        "not have yet a 'latestEnhancedStatus' field and a 'enhancedStatusLastUpdated' field and" when {
+          "has a 'SubmissionRequest' action which does not have yet a 'notifications' field" in {
+            prepareNotificationCollection(parsedNotifications :+ parsedNotification) mustBe true
+            runTest(submission1BeforeMigration, submission1AfterMigration)
+          }
+        }
+      }
+    }
+
+    "update a Submission document" when {
+      "the document has a 'mrn' field and" when {
+        "not have yet a 'latestEnhancedStatus' field and a 'enhancedStatusLastUpdated' field and" when {
+          "the document has a 'CancellationRequest' action and" when {
+            "has a 'SubmissionRequest' action which does not have yet a 'notifications' field" in {
+              prepareNotificationCollection(parsedNotifications) mustBe true
+              runTest(submission2BeforeMigration, submission2AfterMigration)
+            }
+          }
+        }
+      }
+    }
```

Comment:
Duplication of the first 3 levels of these two blocks. For ease of reading they should be merged into one block with two  different inner `in` blocks

Reply:
Good spot :+1: